16.0 mig commown factor out commown administrative docs

Honeyxilia opened the pull request.

Honeyxilia committed:
Adapted to renamed commown_res_partner_sms method

Honeyxilia committed:
Added batch creation support for res.partner overload

Honeyxilia committed:
[test] Replaced depreciated WSGI entrypoint/response wrap

Honeyxilia committed:
[test] Replaced setUp by setUpClass

Honeyxilia committed:
[test] Changed expected web login response/redirect

fcayre committed:
Declare python-magic dependency in the commown_administrative_docs module

Honeyxilia committed:
[test] Revamped form values reading in HTML page

Honeyxilia committed:
Create commown_administrative_docs module

Honeyxilia committed:
Moved code from commown related to admin. documents handling

Honeyxilia committed:
Simplified dependency for partner form view

Honeyxilia committed:
Modified portal detail fields in controller lists

Instead of hardcoding all of the mandatory and optional fields in the selected lists,
we re-define the class lists based on the initial lists from the parent classes.

Honeyxilia committed:
Moved embedded Javascript code in dedicated file

Honeyxilia committed:
Modified file attribue check in account details endpoint

Beforehand, the controller tried to access the filename attribue of the given object. However, if the given variable does not possess a filetype attribute (wrong type of variable), this might lead to an AttribueError being thrown.
As such, we add an operation checking whether the given value truly has a filename attribue and return that value if it exists.

Honeyxilia committed:
[test] Modified pre-requisite and end checks in controller test

- To robustify the test, we demonstrate that the id_card1 field of the modified partner is empty (falsy), to show that the controller call wrote into that field.
- It's not necessary to declare a new environnement to check the end value, as we can simply invalidate the record's cache to force a reaccess to the now updated values.

Honeyxilia committed:
Replaced image resize function by 16.0 implementation

Note: The new resize method in odoo.tools.image does not size up an image
if it's smaller than the provided resolution, as explained by the following documentation :

        "The image is never resized above the current image size. This method is
        only to create a smaller version of the image."

Honeyxilia committed:
[test] Added tests for file size handling

Honeyxilia committed:
[test] Removed useless /my/home redirect value

The account endpoint redirects by default to the /my/home page - as such, providing this link does not serve any purposes

Honeyxilia committed:
[test] Added class for res.partner tests + a test on an invalid filetype

This allows us to test res.partner methods with more granularity.

Honeyxilia committed:
[test] Added test for image upload as a document

fcayre commented on commown_administrative_docs/controllers/details.py:
```diff
@@ -39,7 +39,7 @@ def account(self, redirect=None, **post):
             for field in partner.auto_widget_binary_fields:
                 if post.get(field):
-                    if not post[field].filename:
+                    if not (hasattr(post[field], "filename") and post[field].filename):
```
Peut s'écrire plus simplement `if not getattr(post[field], "filename", False)` je pense.

Honeyxilia committed:
[test] Improve cov.: Changed passed id_card2 value for account details

In the account controller, in order to test if the values we pass are valid files, we check whether they have a filename attribute.
If not, then they are replaced by a False value.

So, we use a deliberatly invalid value for id_card2, in order to trigger this detection and increase coverage.

Honeyxilia commented:
/ocabot merge patch
